drivers/clock_control/clock_control_numaker_scc.c

```diff
@@ -0,0 +1,162 @@
+/*
+ * Copyright (c) 2023 Nuvoton Technology Corporation.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT nuvoton_numaker_scc
+
+#include <soc.h>
+#include <zephyr/drivers/clock_control.h>
+#include <zephyr/drivers/clock_control/clock_control_numaker.h>
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(clock_control_numaker_scc, CONFIG_CLOCK_CONTROL_LOG_LEVEL);
+
+struct numaker_scc_config {
+	uint32_t clk_base;
+	int hxt;
+	int lxt;
+	int hirc48;
+	uint32_t clk_pclkdiv;
+	uint32_t core_clock;
+};
+
+static inline int numaker_scc_on(const struct device *dev, clock_control_subsys_t subsys)
+{
+	ARG_UNUSED(dev);
+
+	struct numaker_scc_subsys *scc_subsys = (struct numaker_scc_subsys *)subsys;
+
+	if (scc_subsys->subsys_id == NUMAKER_SCC_SUBSYS_ID_PCC) {
+		SYS_UnlockReg();
+		CLK_EnableModuleClock(scc_subsys->pcc.clk_modidx);
+		SYS_LockReg();
+	} else {
+		return -EINVAL;
```

Comment:
I'd suggest inverting the logic here, change to

```
if (scc_subsys->subsys_id != NUMAKER_SCC_SUBSYS_ID_PCC) {
        return -EINVAL;
}
```

and let the normal execution flow with no indentation. Same below, no big deal in this case but generally makes code easier to follow.

Reply:
This code structure is reserved to support future subsys_id definitions if need be.
for ex: NUMAKER_SCC_SUBSYS_ID_PCC2, NUMAKER_SCC_SUBSYS_ID_PCC3